### ISSUE TITLE
java.io.File in ring response :body is never sent by SSL server

### ISSUE THREAD

**Issue by cemerick** [ztellman/aleph#245]
Aleph/ring/netty responses that carry a `File` `:body` are never "committed", i.e. response headers (including `Content-Length`) are sent, but the body never is.

```clojure
(aleph.http/start-server
  (fn [_] {:status 200 :headers {} :body (java.io.File. "project.clj")})
  {:port 8089
   :ssl-context (aleph.netty/self-signed-ssl-context)})
```

```
$ curl -vk https://localhost:8089
......
> GET / HTTP/1.1
> User-Agent: curl/7.35.0
> Host: localhost:8089
> Accept: */*
> 
< HTTP/1.1 200 OK
* Server Aleph/0.4.1 is not blacklisted
< Server: Aleph/0.4.1
< Connection: Keep-Alive
< Date: Mon, 02 May 2016 14:30:39 GMT
< content-length: 672
```

The connection blocks there, but doesn't close.

Normal behaviour is restored by either not using SSL, or returning `InputStream` bodies instead. e.g. this works:

```clojure
(aleph.http/start-server
  (fn [_] {:status 200 :headers {} :body (java.io.FileInputStream. "project.clj")})
  {:port 8089
   :ssl-context (aleph.netty/self-signed-ssl-context)})
```

**Comment by ztellman** [comment]
Crap, yeah, zero copy writes don't work there, and apparently Netty doesn't
handle that for me. Should be a straightforward fix, I'll update once it's
done.

**Comment by ztellman** [comment]
This is fixed in `0.4.2-alpha3`.

**Comment by cemerick** [comment]
Confirmed, thanks!
